src/Raven.Client/Http/HttpCache.cs

```diff
@@ -76,51 +87,48 @@ public void Release()
 
             public void Dispose()
             {
-                Release();
+                ReleaseRef();
             }
 
             ~HttpCacheItem()
             {
-                try
-                {
-                    Release();
-                }
-                catch (ObjectDisposedException)
-                {
-                    // nothing that can be done here
-                }
+#if !RELEASE
+                // Hitting this on DEBUG and/or VALIDATE and getting a higher number than 0 means we have a leak.
+                // On release we will leak, but wont crash. 
+                if (_usages > 0)
+                    throw new Exception("Detected a leak on HttpCache when running the finalizer.");
```

Comment:
Better exception, also add the item url so we'll have something to help us track down the cause.

Reply:
Couldnt find any that makes sense, do you have any particular one in mind?

Reply:
Low memory exception? BAsically I just hate the bare exception stuff.